src/System.Runtime/tests/Performance/Perf.StringBuilder.cs

```diff
@@ -144,5 +144,47 @@ public static void StringBuilderToString()
                 }
             }
         }
+
+        [Benchmark]
+        [InlineData(20)]
+        [InlineData(200)]
+        [InlineData(1000)]
+        public void AppendMemoryAsMemory(int length)
+        {
+            PerfUtils utils = new PerfUtils();
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                // Setup - Create a string of the specified length
+                string builtString = utils.CreateString(length);
+                ReadOnlyMemory<char> memory = builtString.AsMemory();
+                StringBuilder empty = new StringBuilder();
+
+                // Actual perf testing
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < 10000; i++)
+                        empty.Append(memory); // Appends a string of length "length" to an increasingly large StringBuilder
+            }
+        }
+
+        [Benchmark]
+        [InlineData(20)]
+        [InlineData(200)]
+        [InlineData(1000)]
+        public void AppendMemoryAsObject(int length)
+        {
+            PerfUtils utils = new PerfUtils();
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                // Setup - Create a string of the specified length
+                string builtString = utils.CreateString(length);
+                ReadOnlyMemory<char> memory = builtString.AsMemory();
+                StringBuilder empty = new StringBuilder();
+
+                // Actual perf testing
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < 10000; i++)
+                        empty.Append((object)memory); // cast to object deliberately forces memory.ToString() to be called so this can be compared to AppendAsReadOnlyMemory
```

Comment:
Wouldn't this cast result in an overhead that we don't want to include in the measurements? Should we cache it into a local outside of the measurement loop?

Reply:
Changed the assignment outside the loop to use an object type and moved the explanatory comment up to that location.